lib/measured/unit.rb

```diff
@@ -1,14 +1,33 @@
 class Measured::Unit
   include Comparable
 
-  def initialize(name, aliases: [], value: nil)
+  def initialize(name, aliases: [], case_sensitive: false, value: nil)
     @name = name.to_s
     @names = ([@name] + aliases.map{|n| n.to_s }).sort
 
+    @case_sensitive = case_sensitive
     @conversion_amount, @conversion_unit = parse_value(value) if value
   end
 
-  attr_reader :name, :names, :conversion_amount, :conversion_unit
+  attr_reader :name, :names, :case_sensitive, :conversion_amount, :conversion_unit
+
+  def name_eql?(name_to_compare)
+    name = with_case_sensitivity self.name
+    name_to_compare = with_case_sensitivity name_to_compare
+
+    name.eql? name_to_compare.to_s
+  end
+
+  def names_include?(name_to_compare)
+    current_names = with_case_sensitivity self.names
+    name_to_compare = with_case_sensitivity name_to_compare
+
+    current_names.include? name_to_compare.to_s
+  end
+
+  def add_alias(aliases)
+    @names = (@names << aliases).flatten.sort
```

Comment:
It's used in Measured::Conversion, it allows Measured::Unit to handle each check for #unit? and #unit_for_alias? . You are completely right though, after I created #with_case_sensitivity I forgot I could clean up these methods as well